src/main/java/org/folio/service/inventory/InventoryInstanceManager.java

```diff
@@ -346,18 +342,51 @@ private Future<String> createInstance(JsonObject instanceRecJson, RequestContext
 
   public Future<CompositePoLine> openOrderHandleInstance(CompositePoLine compPOL, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
     return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
-      .compose(consortiumConfiguration -> {
+      .compose(configuration -> {
+        Future<String> instanceIdFuture;
         if (compPOL.getInstanceId() != null) {
-          return consortiumConfiguration.isEmpty() ? Future.succeededFuture(compPOL.getInstanceId())
-            : shareInstanceAmongTenantsIfNeeded(compPOL.getInstanceId(), consortiumConfiguration.get(), compPOL.getLocations(), requestContext);
+          instanceIdFuture = Future.succeededFuture(compPOL.getInstanceId());
         } else {
-          return getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext)
-            .compose(instanceId -> consortiumConfiguration.isEmpty() ? Future.succeededFuture(instanceId)
-              : shareInstanceAmongTenantsIfNeeded(instanceId, consortiumConfiguration.get(), compPOL.getLocations(), requestContext));
+          instanceIdFuture = getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext);
         }
+        configuration.ifPresent(
+          config -> instanceIdFuture
+            .compose(instanceId -> shareInstanceAmongTenantsIfNeeded(instanceId, config, compPOL.getLocations(), requestContext))
+        );
+        return instanceIdFuture;
```

Comment:
here only instanceIdFuture is returned, compose to share instance is not returned.
```
  return configuration.isEmpty() ? instanceIdFuture : instanceIdFuture
    .compose(instanceId -> shareInstanceAmongTenantsIfNeeded(instanceId, configuration.get(), compPOL.getLocations(), requestContext));
```